app/xds_api/views.py

```diff
@@ -174,3 +178,175 @@ def post(self, request, *args, **kwargs):
+                if idx == len(coursesDict) - 1:
+                    courseQuery += metadata_key_hash
+                else:
+                    courseQuery += (metadata_key_hash + ",")
+
+            if (len(coursesDict) > 0):
+                # get search string
+                composite_api_url = XDSConfiguration.objects.first()\
+                    .target_xis_metadata_api
+                api_url = composite_api_url + courseQuery
+
+                # make API call
+                response = get_request(api_url)
+                responseJSON = json.dumps(response.json())
+
+                if (response.status_code == 200):
+                    formattedResponse = metadata_to_target(responseJSON)
+                    interestList['courses'] = formattedResponse
+
+                    return Response(interestList,
+                                    status=status.HTTP_200_OK)
+                else:
+                    return Response(response.json(),
+                                    status=status.HTTP_503_SERVICE_UNAVAILABLE)
+        elif request.method == 'PATCH':
+            # Assign data from request to serializer
+            user = request.user
+
+            # check user is logged in
+            if not request.user.is_authenticated:
+                return Response({'Please login to update Interest List'},
+                                status.HTTP_401_UNAUTHORIZED)
+
+            # check user is owner of list
+            if not request.user == queryset.owner:
+                return Response({'Current user does not have access to modify '
+                                 'the list'},
+                                status.HTTP_401_UNAUTHORIZED)
+            # save new courses)
+            save_courses(request.data['courses'])
+
+            serializer = InterestListSerializer(queryset, data=request.data)
+
+            if not serializer.is_valid():
+                # If not received send error and bad request status
+                logger.info(json.dumps(request.data))
+                return Response(serializer.errors,
+                                status=status.HTTP_400_BAD_REQUEST)
+
+            serializer.save(owner=user)
+
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+    except HTTPError as http_err:
+        logger.error(http_err)
+        return Response(errorMsg, status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except ObjectDoesNotExist as not_found_err:
+        logger.error(not_found_err)
+        return Response(errorMsg, status.HTTP_404_NOT_FOUND)
+    except Exception as err:
+        logger.error(err)
+        return Response(errorMsg, status.HTTP_500_INTERNAL_SERVER_ERROR)
+    else:
+        return Response(serializer_class.data, status.HTTP_200_OK)
+
+
+@api_view(['POST'])
+def add_course_to_lists(request):
+    """This method handles request for adding a single course to multiple
```

Comment:
Do we have a scenario where we would add a single course to multiple lists at once?  I like the idea it's similar to how YouTube playlists work.